core/domain/fs_domain.py

```diff
@@ -248,20 +256,20 @@ def listdir(self, dir_name):
 class AbstractFileSystem:
     """Interface for a file system."""
 
-    def __init__(self, impl):
+    def __init__(self, impl: AbstractFileSystem) -> None:
         """Constructs a AbstractFileSystem object."""
         self._impl = impl
 
     @property
-    def impl(self):
+    def impl(self) -> AbstractFileSystem:
```

Comment:
```suggestion
    def impl(self) -> GeneralFileSystem:
```

Reply:
![image](https://user-images.githubusercontent.com/37413698/151685537-1394fd8f-d0a9-4781-8317-50fe828f377c.png)

Reply:
You need to also modify in in the __init__ above